x-pack/plugins/stack_connectors/public/connector_types/servicenow_itsm/servicenow_itsm_params.tsx

```diff
@@ -208,16 +257,43 @@ const ServiceNowParamsFields: React.FunctionComponent<
         index
       );
     }
+
+    if (
+      (isTestResolveAction || isTestTriggerAction) &&
+      (!isEmpty(actionParams.subActionParams?.incident) ||
+        actionParams.subActionParams?.comments?.length)
+    ) {
+      editAction('subActionParams', { incident: {}, comments: undefined }, index);
+      return;
+    }
+
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [actionConnector]);
+  }, [actionConnector, isTestResolveAction, isTestTriggerAction]);
 
   return (
     <>
-      <EuiTitle size="s">
-        <h3>{i18n.INCIDENT}</h3>
-      </EuiTitle>
+      {executionMode === ActionConnectorMode.Test ? (
+        <EuiFormRow fullWidth label={i18n.EVENT_ACTION_LABEL}>
+          <EuiSelect
+            fullWidth
+            data-test-subj="eventActionSelect"
+            options={eventActionOptions}
+            value={eventAction}
+            onChange={(e) => handleEventActionChange(e.target.value as EventAction)}
+          />
+        </EuiFormRow>
+      ) : null}
       <EuiSpacer size="m" />
-      {selectedActionGroupId !== ACTION_GROUP_RECOVERED ? (
+      {(executionMode !== ActionConnectorMode.Test ||
```

Comment:
Now that we default the `eventAction` do we still need this? It seems that the `Incident` label is shown in all cases.